core/src/main/scala/akka/persistence/cassandra/journal/TagWriter.scala

```diff
@@ -237,6 +236,12 @@ import scala.concurrent.duration._
       sender() ! ResetPersistenceIdComplete
   }
 
+  private def sendPubsubNotification(): Unit = {
+    pubsub.foreach {
+      _ ! DistributedPubSubMediator.Publish("akka.persistence.cassandra.journal.tag", tag)
+    }
+  }
```

Comment:
hmm, there is one thing that I should improve here. It should be delayed so that the notification doesn't arrive before the eventual-consistency-delay because then the query will be fruitless anyway. I think that was done on the receiving side in 0.5 branch, but now we also have the flush-interval on the write side, so maybe better to add the delay on the sending side.

```
delay = eventual-consistency-delay - flush-interval
```

Reply:
:+1:

Reply:
It is still delayed on the receiving side: https://github.com/akka/akka-persistence-cassandra/blob/master/core/src/main/scala/akka/persistence/cassandra/query/EventsByTagStage.scala#L272

Reply:
Strike that, `flush-interval` is before writing so not part of this equation.
EventsByTagStage is already scheduling the `TagNotification` by taking the eventual-consistency-delay into account.

Reply:
`delay = eventual-consistency-delay - flush-interval` isn't strictly right as the flushed events will have timestamps up from the current time to current time - flush-interval. But given the write to the database will take some time and that flush interval is typically a small duration maybe that is good enough?

Reply:
Snap, let's leave it as is

Reply:
right, the timestamps are from before the flush-interval, before the writes.

I think it's fine as it is to schedule the TagNotification in the EventsByTagStage.